File: Src/python/trigger.py
import bytes

instanceMap = {}

class TriggerInstance:
    def __init__(self, signature):
        genID = uuid.getnode()
        instanceMap[genID] = signature
        self.id = hex(genID)

def getSignature(instance):
    return instanceMap[str(instance.id)]

def removeInstance(instance):
    del instanceMap[str(instance.id)]

File: Src/javascript/trigger.js
const uuid = require('uuid')

instanceMap = {}

class TriggerInstance {
    constructor(signature) {
        genID = uuid.v4()
        instanceMap[genID] = signature
        this.id = uuid.parse(genID)
    }
}

getSignature = (instance) => {
    return instanceMap[uuid.stringify(instance.id)]
}

removeInstance = (instance) => {
    instanceMap[uuid.stringify(instance.id)] = undefined
}